### PR TITLE
Extract RAML and OpenAPI schema parsers to their own package

### LINKED ISSUE #48: ramldt2jsonschema package has link to fs module
The 'ramldt2jsonschema' module you are using has a link to the 'fs' module in the code. As a result, it turned out to be impossible to use parser version 0.14 with the asyncapi-react project when it is built only for the browser.

Is there anything you can do about this?

### PR DESCRIPTION
**Pull request by fmvilas**
It makes the bundle size of the parser much bigger. Next one could be the OpenAPI parser too.

The new package is here: https://www.github.com/asyncapi/raml-dt-schema-parser/

fixes #48

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -19,8 +18,4 @@ parser.registerSchemaParser([
   'application/vnd.oai.openapi+yaml;version=3.0.0',
 ], openapiSchemaParser);
 
-parser.registerSchemaParser([
-  'application/raml+yaml;version=1.0',
-], ramlDtParser);
-
```
Comment (jonaslagoni) [inline lib/index.js]: I can understand why we move the functionality for parsing RAML. However why do we remove the parser completely :thinking: ?

Comment (derberg) [reply lib/index.js]: @jonaslagoni this is optional parser, so doesn't have to be here, it actually makes a lot sense to have all the additional parsers used as plugins. Look on readme here how you would use it https://github.com/asyncapi/raml-dt-schema-parser/

This change makes the parser very nice as you can decide not to use our parser for RAML but write your own and plug it in. Plus, the bundle size is getting smaller as you have in it only the required minimum

@fmvilas it shoudl definitely happen with openapi parser too, please do not forget to create issue.

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   "author": "Fran Mendez <fmvilas@gmail.com> (fmvilas.com)",
   "license": "Apache-2.0",
   "devDependencies": {
+    "@asyncapi/raml-dt-schema-parser": "^1.0.4",
```
Comment (derberg) [inline package.json]: why do we still test the RAML parser here and not https://github.com/asyncapi/raml-dt-schema-parser/ ?

File: lib/index.js (hunk 1 above)
Comment (jonaslagoni) [reply lib/index.js]: Alright understandable, do we have/need an issue in the generator which includes this new parser 🤔?

File: package.json (hunk 2 above)
Comment (jonaslagoni) [reply package.json]: I actually don't mind it being testing here, great to have some integration tests. I guess we could actually duplicate this test in the RAML parser as well?

Comment (derberg) [reply package.json]: maybe as a short term solution, but ideally there should be a dummy parser there in the repo used for testing that also could be used as a reference in docs. So we at least need to have a followup issue for that to improve

Comment (jonaslagoni) [reply package.json]: :thinking: Yea you are right, would probably be best to use a dummy parser here and then in the RAML parser add this test :)

File: lib/index.js (hunk 1 above)
Comment (fmvilas) [reply lib/index.js]: What new parser, @jonaslagoni?

File: package.json (hunk 2 above)
Comment (fmvilas) [reply package.json]: Good point. Moving the test.

File: lib/index.js (hunk 1 above)
Comment (jonaslagoni) [reply lib/index.js]: The optional RAML Schema parser that is being separated, I would imagine we had to include that [here ](https://github.com/asyncapi/generator/blob/master/lib/generator.js#L209) right?

File: package.json (hunk 2 above)
Comment (fmvilas) [reply package.json]: @derberg @jonaslagoni https://github.com/asyncapi/raml-dt-schema-parser/pull/2

File: lib/index.js (hunk 1 above)
Comment (fmvilas) [reply lib/index.js]: Oh, good point! You're right. Doing it now. Thanks, mate!

File: package.json (hunk 2 above)
Comment (fmvilas) [reply package.json]: Ok, finished moving tests (for OpenAPI too).

File: lib/index.js (hunk 1 above)
Comment (fmvilas) [reply lib/index.js]: Took the opportunity to extract the OpenAPI schema parser too: github.com/asyncapi/openapi-schema-parser.

**Comment by derberg** [comment]
@fmvilas oh man, you already did OpenAPI plugin extraction. Well done!

File: lib/index.js (hunk 1 above)
Comment (fmvilas) [reply lib/index.js]: @jonaslagoni https://github.com/asyncapi/generator/pull/247 😄

**Comment by fmvilas** [comment]
@jonaslagoni do you think it's fine now?

**Pull request merged**